[rules_cc] Recognize CC as GCC when applicable

Running `cc -v` when CC is really GCC under the hood will still print GCC
information, similar to the logic with Clang immediately above.

--- cc/private/toolchain/unix_cc_configure.bzl
+++ cc/private/toolchain/unix_cc_configure.bzl
@@ -273,10 +273,7 @@ def _is_clang(repository_ctx, cc):
     return "clang" in repository_ctx.execute([cc, "-v"]).stderr

 def _is_gcc(repository_ctx, cc):
-    # GCC's version output uses the basename of argv[0] as the program name:
-    # https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=gcc/gcc.cc;h=158461167951c1b9540322fb19be6a89d6da07fc;hb=HEAD#l8728
-    cc_stdout = repository_ctx.execute([cc, "--version"]).stdout
-    return cc_stdout.startswith("gcc ") or cc_stdout.startswith("gcc-")
+    return "gcc version " in repository_ctx.execute([cc, "-v"]).stderr.split('\n')[-2]

 def _get_compiler_name(repository_ctx, cc):
     if _is_clang(repository_ctx, cc):
